module-4/metaanalyse/metaanalyse.py

```diff
@@ -0,0 +1,81 @@
+import os
+import csv
+import argparse
+from PyPDF2 import PdfReader
+
+def extract_pdf_metadata(file_path):
+    metadata = {
+        'File Name': os.path.basename(file_path),
+        'Title': 'undefined',
+        'Author': 'undefined',
+        'Creator': 'undefined',
+        'Created': 'undefined',
+        'Modified': 'undefined',
+        'Subject': 'undefined',
+        'Keywords': 'undefined',
+        'Description': 'undefined',
+        'Producer': 'undefined',
+        'PDF Version': 'undefined'
+    }
+    try:
+        reader = PdfReader(file_path)
+        info = reader.metadata
+
+        metadata['Title'] = info.get('/Title', 'undefined') or 'undefined'
+        metadata['Author'] = info.get('/Author', 'undefined') or 'undefined'
+        metadata['Creator'] = info.get('/Creator', 'undefined') or 'undefined'
+        metadata['Created'] = info.get('/CreationDate', 'undefined') or 'undefined'
+        metadata['Modified'] = info.get('/ModDate', 'undefined') or 'undefined'
+        metadata['Subject'] = info.get('/Subject', 'undefined') or 'undefined'
+        metadata['Keywords'] = info.get('/Keywords', 'undefined') or 'undefined'
+        metadata['Description'] = info.get('/Description', 'undefined') or 'undefined'
+        metadata['Producer'] = info.get('/Producer', 'undefined') or 'undefined'
```

Comment:
the second param of the `get` function is optional and will only be returned if the `key` was not found in your dictionary --> the `or 'undefined'` part is unnecessary.